common/common.h

```diff
@@ -32,9 +32,15 @@
 } while(0)
 
 // build info
-// Remove these external build flags as they are injected as preprocessor macro's from the Xcode project
-// extern int LLAMA_BUILD_NUMBER;
-// extern char const *LLAMA_COMMIT;
+// If macro is not defined (passed via Xcode), set the corresponding variable as defined via external variable.
+#ifndef LLAMA_BUILD_NUMBER
+extern int LLAMA_BUILD_NUMBER;
```

Comment:
Where are these externs coming from exactly? Just out of interest?

Reply:
Afaik, they're usually passed via the CI. Locally, they're just dummy values.

Reply:
But, afaik, `extern` is just to signal that the variable is somewhere present in a different translation unit? So the variable needs to be defined somewhere right? And how do the preprocessor macros play into that?
Just wanted to clarify that you don't mix up preprocessor macros and external declarations. Otherwise, this doesn't have great relevance to me.

Reply:
I used the preprocessor macros as a substitution mechanism for compiling the `common.h` header, as the `extern` variables are passed (as far as I'm seeing) via some CMake mechanism (that we obviously don't use when compiling this thing as an XCFramework).